frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -135,6 +136,58 @@ export const getJoinedCardsDataset = (
   return Array.from(groupedData.values());
 };
 
+export const applySquareRootScaling = (value: RowValue): RowValue => {
+  if (typeof value === "number") {
+    return Math.sqrt(value);
+  }
+
+  return value;
+};
+
+export const getTransformedDataset = (
+  dataset: Record<DataKey, RowValue>[],
+  seriesModels: SeriesModel[],
+  settings: ComputedVisualizationSettings,
+  dimensionModel: DimensionModel,
+) => {
+  const seriesDataKeys = seriesModels.map(seriesModel => seriesModel.dataKey);
+  const seriesDataKeysSet = new Set(seriesDataKeys);
+
+  let transformedDataset = replaceValues(
+    dataset,
+    getNullReplacerFunction(settings, seriesModels),
+  );
+
+  if (settings["stackable.stack_type"] === "normalized") {
+    transformedDataset = getNormalizedDataset(
+      transformedDataset,
+      seriesDataKeys,
+      dimensionModel.dataKey,
+    );
+  }
+
+  if (settings["graph.y_axis.scale"] === "pow") {
+    transformedDataset = replaceValues(
+      transformedDataset,
+      (dataKey: string, value: RowValue) =>
+        seriesDataKeysSet.has(dataKey) ? applySquareRootScaling(value) : value,
+    );
+  }
+
+  return transformedDataset;
+};
+
+export const getMetricDisplayValueGetter = (
```

Comment:
Should we maybe move this to the `option` directory instead, since it isn't actually used for any of the chart model code?

Maybe we can have a file `echarts/cartesian/option/utils.ts`